src/Servers/HttpSys/src/UrlPrefixCollection.cs

```diff
@@ -138,12 +146,56 @@ internal void RegisterAllPrefixes(UrlGroup urlGroup)
             {
                 _urlGroup = urlGroup;
                 // go through the uri list and register for each one of them
-                foreach (var pair in _prefixes)
+                // Call ToList to avoid modification when enumerating.
+                foreach (var pair in _prefixes.ToList())
+                {
+                    var urlPrefix = pair.Value;
+                    if (urlPrefix.PortValue == 0)
+                    {
+                        if (urlPrefix.IsHttps)
+                        {
+                            throw new InvalidOperationException("Cannot bind to port 0 with https.");
+                        }
+
+                        if (!FindHttpPortUnsynchronized(pair.Key, urlPrefix))
+                        {
+                            throw new HttpSysException(Marshal.GetLastWin32Error(), "Could not bind to port 0.");
```

Comment:
It's unreliable to re-retrieve the error code later, it may have changed. Flow the original exception back here as an out param?